app/src/androidTest/java/com/github/se/project/ui/endToEnd/TutorEndToEnd.kt

```diff
@@ -0,0 +1,251 @@
+
+  @Before
+  fun setUp() {
+    context = mock(Context::class.java)
+    whenever(mockProfileRepository.addProfile(any(), any(), any())).thenAnswer { invocation ->
+      val onSuccess = invocation.arguments[1] as () -> Unit
+      onSuccess() // Simulate a successful update
+    }
+    whenever(mockProfileRepository.getProfiles(any(), any())).thenAnswer { invocation ->
+      val onSuccess = invocation.arguments[0] as (List<Profile>) -> Unit
+      onSuccess(listOf(mockStudent)) // Simulate a list of profiles with our beloved Ozymandias
+    }
+    whenever(mockProfileRepository.getNewUid()).thenReturn("mockUid")
+    whenever(mockLessonRepository.updateLesson(any(), any(), any())).thenAnswer { invocation ->
+      mockLessons = listOf(invocation.arguments[0] as Lesson)
+      val onSuccess = invocation.arguments[1] as () -> Unit
+      onSuccess()
+    }
+    whenever(mockLessonRepository.getLessonsForTutor(eq("mockUid"), any(), any())).thenAnswer {
+        invocation ->
+      val onSuccess = invocation.arguments[1] as (List<Lesson>) -> Unit
+      onSuccess(mockLessons)
+    }
+    whenever(mockLessonRepository.getAllRequestedLessons(any(), any())).thenAnswer { invocation ->
+      val onSuccess = invocation.arguments[0] as (List<Lesson>) -> Unit
+      onSuccess(mockLessons)
+    }
+    whenever(mockLessonRepository.updateLesson(any(), any(), any())).thenAnswer { invocation ->
+      mockLessons = listOf(invocation.arguments[0] as Lesson)
+      val onSuccess = invocation.arguments[1] as () -> Unit
+      onSuccess() // Simulate a successful update
+    }
+    val mockLessonFlow = MutableStateFlow<Lesson?>(mockLessons[0])
+    doReturn(mockLessonFlow).`when`(mockLessonViewModel).selectedLesson
+    mockLessonFlow.value = mockLessons[0]
+  }
+
+  @Test
+  fun TutorEndToEndTest() {
+    composeTestRule.setContent {
+      PocketTutorApp(true, viewModel(), mockProfileViewModel, mockLessonViewModel)
+    }
+    // Thread.sleep(50000)
+    // Sign In Screen
+    composeTestRule.onNodeWithTag("logo").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("loginButton").performClick()
+    composeTestRule.onNodeWithTag("firstNameField").assertIsDisplayed()
+
+    // Create Profile Screen
+    composeTestRule.onNodeWithTag("firstNameField").performTextInput("John")
+    composeTestRule.onNodeWithTag("lastNameField").performTextInput("Doe")
+    composeTestRule.onNodeWithTag("phoneNumberField").performTextInput("0213456789")
+    composeTestRule.onNodeWithTag("roleButtonTutor").performClick()
+    composeTestRule.onNodeWithTag("sectionDropdown").performClick()
+    composeTestRule.onNodeWithTag("sectionDropdownItem-SC").performClick()
+    composeTestRule.onNodeWithTag("academicLevelDropdown").performClick()
+    composeTestRule.onNodeWithTag("academicLevelDropdownItem-BA3").performClick()
+    composeTestRule.onNodeWithTag("confirmButton").performClick()
+    assertEquals(Role.TUTOR, mockProfileViewModel.currentProfile.value?.role)
```

Comment:
Good way to check profile creation was successful. Other fields could be tested if we wanted to be thorough, although that is not strictly necessary.